iModelCore/ecobjects/src/SchemaMerger.cpp

```diff
@@ -218,8 +236,30 @@ BentleyStatus SchemaMerger::MergeSchemas(SchemaMergeResult& result, bvector<ECSc
     //Make a copy of the input vectors so we don't modify the original (given to us as const anyways)
     bvector<ECSchemaCP> left(rawLeft);
     bvector<ECSchemaCP> right(rawRight);
-    ECSchema::SortSchemasInDependencyOrder(left);
-    ECSchema::SortSchemasInDependencyOrder(right);
+    auto doNotMergeReferences = options.DoNotMergeReferences();
+    if(!doNotMergeReferences)
+        {
+        ECSchema::SortSchemasInDependencyOrder(left);

```

Comment:
I believe we should also sort the provided schemas even if the flag is false.
Better add a new boolean parameter to SortSchemasInDependencyOrder which gets passes along to BuildDependencyOrderedSchemaList() and indicates whether referenced schemas should be included in that list or not